### PR TITLE
Lindon/28 refactor accumulator

### PR DESCRIPTION
**Pull request by michaellindon**
I added running computation of p_value, bayes_factor, posterior_probability and log_bayes_factor.
This is much more convenient for the ssrm-service because the previous functions such as sequential_p_values and sequential_posteriors were better suited to offline batch analysis.

### CODE REVIEW

File: ssrm_test/test_ssrm_test.py [hunk 1]
```diff
@@ -226,3 +228,22 @@ def test_data_validator():
     assert _validate_data(np.array([[1, 3], [4, 3]]))
     assert not _validate_data(np.array([[1, 3], [4, 3.5]]))
     assert not _validate_data(np.array([[1, 3], [4, "hello"]]))
+
+
+def test_regression():
```
[the remark this answers is not in the document]
Comment (michaellindon) [reply ssrm_test/test_ssrm_test.py:233]: Not quite sure what the name of those tests are where you check that computation always produces the same numeric result which is accepted to be true. Can you suggest an appropriate name?

File: ssrm_test/ssrm_test.py [hunk 2]
```diff
@@ -83,11 +83,25 @@ def accumulator(acc: dict, new_data_point: np.ndarray) -> dict:
...
+        warnings.simplefilter("ignore")
+        bayes_factor = np.exp(log_bayes_factor)
+    post_prob = posterior_probability(bayes_factor)
+    p_value = min(1 / bayes_factor, acc["p_value"])
     out = {
-        "log_marginal_likelihood_M1": acc["log_marginal_likelihood_M1"]
-        + log_posterior_predictive(new_data_point, acc["posterior_M1"]),
-        "log_marginal_likelihood_M0": acc["log_marginal_likelihood_M0"]
-        + multinomiallogpmf(new_data_point, new_data_point.sum(), acc["posterior_M0"])
+        "log_bayes_factor": log_bayes_factor,
```
[the remark this answers is not in the document]
Comment (michaellindon) [reply ssrm_test/ssrm_test.py]: done

File: ssrm_test/test_ssrm_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajithopti) [reply ssrm_test/test_ssrm_test.py:242]: Perhaps, `test_sequential_stats_values`?

File: ssrm_test/ssrm_test.py [hunk 3]
```diff
@@ -83,15 +85,34 @@ def accumulator(acc: dict, new_data_point: np.ndarray) -> dict:
...
+        bayes_factor = np.exp(log_bayes_factor)
+        posterior_odds = np.exp(log_posterior_odds)
+    post_prob = posterior_probability(posterior_odds)
+    p_value = min(1 / bayes_factor, acc[P_VALUE])
     out = {
-        "log_marginal_likelihood_M1": acc["log_marginal_likelihood_M1"]
-        + log_posterior_predictive(new_data_point, acc["posterior_M1"]),
-        "log_marginal_likelihood_M0": acc["log_marginal_likelihood_M0"]
-        + multinomiallogpmf(new_data_point, new_data_point.sum(), acc["posterior_M0"])
+        LOG_POSTERIOR_ODDS: log_posterior_odds,
```
Comment (ajithopti) [inline ssrm_test/ssrm_test.py:104]: Please make any pertinent updates to `RELEASE.md`, under a `## [Unreleased]` section above any other releases.

File: .flake8 [hunk 4]
```diff
@@ -1,2 +1,3 @@
 [flake8]
 max-line-length = 120
+ignore = F405
```
Comment (ajithopti) [inline .flake8:3]: You'll need to add `F403,W503` to ignore.

**Pull request merged**